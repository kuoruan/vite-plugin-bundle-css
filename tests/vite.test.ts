import type { RollupOutput } from "rollup";
import { build, mergeConfig, type UserConfig } from "vite";

import bundleCss from "@/index";

import { getAssetOutputs } from "./utils";
import viteConfig from "./vite.config";

describe("vite", () => {
  it("should build", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        lib: {
          entry: "/fixtures/css.ts",
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [bundleCss()],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it("should build with import mode", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        cssCodeSplit: true,
        lib: {
          entry: "/fixtures/css.ts",
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [
        bundleCss({
          mode: "import",
        }),
      ],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it("should build with fileName", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        lib: {
          entry: "/fixtures/css.ts",
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [
        bundleCss({
          fileName: "assets/bundle.css",
        }),
      ],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it("should build with filter", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        lib: {
          entry: "/fixtures/scss.ts",
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [
        bundleCss({
          include: ["**/*.js"],
        }),
      ],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it("should build with transform", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        lib: {
          entry: "/fixtures/css.ts",
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [
        bundleCss({
          transform: (code) => {
            return code.replaceAll("color", "background-color");
          },
        }),
      ],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it("should build with filter in import mode", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        cssCodeSplit: true,
        lib: {
          entry: ["/fixtures/scss.ts", "/fixtures/css.ts"],
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [
        bundleCss({
          include: ["**/css.css"],
          mode: "import",
        }),
      ],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it("should build with multiple entries", async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        cssCodeSplit: true,
        lib: {
          entry: ["/fixtures/css.ts", "/fixtures/scss.ts"],
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [bundleCss()],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });

  it('should build with "import" mode in multiple entries', async () => {
    const config = mergeConfig<UserConfig, UserConfig>(viteConfig, {
      build: {
        cssCodeSplit: true,
        lib: {
          entry: ["/fixtures/css.ts", "/fixtures/scss.ts"],
          formats: ["es"],
          fileName: "[name].js",
        },
      },
      plugins: [
        bundleCss({
          mode: "import",
        }),
      ],
    });

    const data = (await build(config)) as RollupOutput[];

    expect(getAssetOutputs(data)).toMatchSnapshot();
  });
});
