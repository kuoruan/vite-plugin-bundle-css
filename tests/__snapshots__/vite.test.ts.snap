// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite > should build 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b {
  color: red;
}
.b .c {
  color: green;
}
",
    "type": "asset",
  },
  {
    "fileName": "style.css",
    "name": "style.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with "import" mode in multiple entries 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": "@import "./css.css";
@import "./scss.css";
",
    "type": "asset",
  },
  {
    "fileName": "css.css",
    "name": "css.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
  {
    "fileName": "scss.css",
    "name": "scss.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".c .d{color:red}.d .e{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with fileName 1`] = `
[
  {
    "fileName": "assets/bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b {
  color: red;
}
.b .c {
  color: green;
}
",
    "type": "asset",
  },
  {
    "fileName": "style.css",
    "name": "style.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with filter 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": "",
    "type": "asset",
  },
  {
    "fileName": "style.css",
    "name": "style.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".c .d{color:red}.d .e{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with filter in import mode 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": "@import "./css.css";
",
    "type": "asset",
  },
  {
    "fileName": "css.css",
    "name": "css.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
  {
    "fileName": "scss.css",
    "name": "scss.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".c .d{color:red}.d .e{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with import mode 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": "@import "./css.css";
",
    "type": "asset",
  },
  {
    "fileName": "css.css",
    "name": "css.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with multiple entries 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b {
  color: red;
}
.b .c {
  color: green;
}
.c .d {
  color: red;
}
.d .e {
  color: green;
}
",
    "type": "asset",
  },
  {
    "fileName": "css.css",
    "name": "css.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
  {
    "fileName": "scss.css",
    "name": "scss.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".c .d{color:red}.d .e{color:green}
",
    "type": "asset",
  },
]
`;

exports[`vite > should build with transform 1`] = `
[
  {
    "fileName": "bundle.css",
    "name": "bundle.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b {
  background-color: red;
}
.b .c {
  background-color: green;
}
",
    "type": "asset",
  },
  {
    "fileName": "style.css",
    "name": "style.css",
    "needsCodeReference": false,
    "originalFileName": null,
    "source": ".a .b{color:red}.b .c{color:green}
",
    "type": "asset",
  },
]
`;
